src/javascript/JContent/EditFrame/Box.jsx

```diff
@@ -190,64 +192,70 @@ export const Box = React.memo(({
-             onClick={onClick}
-             onDoubleClick={onDoubleClick}
+        <header ref={dragWithChecks}
+                className={headerStyles}
+                jahiatype="header" // eslint-disable-line react/no-unknown-property
+                data-current={isCurrent}
+                data-clicked={isClicked}
+                data-hovered={isHeaderHighlighted}
+                data-jahia-id={element.getAttribute('id')}
+                onMouseOver={onMouseOver}
+                onMouseOut={onMouseOut}
+                onClick={onClick}
+                onDoubleClick={onDoubleClick}
         >
-            <div ref={dragWithChecks}
-                 className={clsx(editStyles.enablePointerEvents, styles.header, 'flexRow_nowrap alignCenter', type === 'area' && styles.isArea, (isClicked || isHeaderHighlighted) && styles.isClicked)}
-            >
-                <Checkbox checked={isSelected} data-sel-role="selection-checkbox" onChange={onSelect}/>
-                {node &&
-                    <Bar
-                        isActionsHidden={isActionsHidden}
-                        node={node}
-                        language={language}
-                        displayLanguage={displayLanguage}
-                        width={currentOffset.width}
-                        currentFrameRef={currentFrameRef}
-                        element={element}/>}
-            </div>
-        </div>
+            <Checkbox checked={isSelected} data-sel-role="selection-checkbox" onChange={onSelect}/>
+            {node &&
+                <Bar
+                    isActionsHidden={isActionsHidden}
+                    node={node}
+                    language={language}
+                    displayLanguage={displayLanguage}
+                    width={currentOffset.width}
+                    currentFrameRef={currentFrameRef}
+                    element={element}/>}
+        </header>
     );
 
-    const boxStyle = !isAnythingDragging && (isCurrent || isClicked) && breadcrumbs.length > 0 ? styles.relHeaderAndFooter : styles.relHeader;
+    const boxStyle = !isAnythingDragging && isClicked && breadcrumbs.length > 0 ? styles.withHeaderAndFooter : styles.withHeader;
 
     return (
         <div ref={rootDiv}
              className={clsx(styles.root, isBarAlwaysDisplayed ? styles.alwaysDisplayedZIndex : styles.defaultZIndex)}
              style={currentOffset}
         >
-            <div className={clsx(styles.rel,
-                isHeaderDisplayed ? boxStyle : styles.relNoHeader,
-                (isCurrent) && !(isSelected || isClicked) ? styles.current : '',
-                (isSelected || isClicked) ? styles.selected : '')}
+            <div className={clsx(
+                styles.box,
+                isHeaderDisplayed ? boxStyle : styles.withNoHeader,
+                (isCurrent) && !(isSelected || isClicked) ? styles.boxCurrent : '',
+                (isSelected || isClicked) ? styles.boxSelected : '')}
```

Comment:
ok for now, but wondering if there's a way to avoid adding `!(isSelected || isClicked)` check so `boxCurrent` style is applied then add `boxSelected` style if selected or clicked, through css priority

Reply:
I did the condition in the CSS side with `.boxHovered:not[.boxSelected]` only to apply the hovered style when the element is not selected.